src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,473 @@ const ActorInit En_Heishi1_InitVars = {
+                        if (this->actor.xzDistFromLink < 60.0f) {
+                            this->linkDetected = true;
+                        } else if (this->actor.xzDistFromLink < 70.0f) {
+                            // this case probably exists to detect link making a sidhop or backflip sound
+                            // from slightly further away than the previous 60 unit check
+                            if (player->actor.velocity.y > -4.0f) {
+                                this->linkDetected = true;
+                            }
+                        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A52098.s")
+                        if (this->linkDetected) {
+                            // ! @bug This appears to be a check to make sure that link is standing on flat ground
+                            // before getting caught. However this is an issue for two reasons:
+                            // 1: When doing a backflip or falling from the upper path, links y velocity will reach
+                            // less than -4.0 before even touching the ground.
+                            // 2: There is one frame when landing from a sidehop where you can sidehop again without
+                            // letting y velocity reach -4.0 or less. This enables the player to do frame perfect
+                            // sidehops onto the next screen and prevent getting caught.
+                            if (!(player->actor.velocity.y > -3.9f)) {
+                                this->linkDetected = false;
+                                // this 60 unit height check is so the player doesnt get caught when going on upper path
+                                if (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < 60.0f) {
+                                    func_80078884(NA_SE_SY_FOUND);
+                                    // "Discovered!"
+                                    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 発見！ ☆☆☆☆☆ \n" VT_RST);
+                                    func_8002DF54(globalCtx, &this->actor, 1);
+                                    sPlayerIsCaught = true;
+                                    this->actionFunc = EnHeishi1_SetupMoveToLink;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A5212C.s")
+s32 EnHeishi1_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                               Actor* thisx) {
+    EnHeishi1* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A521FC.s")
+    // turn the guards head to match the direction he is looking
+    if (limbIndex == 16) {
+        rot->x += (s16)this->headAngle;
```

Comment:
```suggestion
        rot->x += this->headAngle;
```

Reply:
this doesnt work without a cast, even though i agree it makes 0 sense. i can post diff in discord if you want